src/Microsoft.ML.Recommender/doc.xml

```diff
@@ -0,0 +1,90 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<doc>
+  <members>
+    
+    <member name="MatrixFactorizationTrainer">
+      <summary>
+        Train a matrix factorization model using <a href='https://www.csie.ntu.edu.tw/~cjlin/libmf/'>LIBMF.
+      </summary>
+      <remarks>
+        The basic idea of matrix factorization is finding two low-rank factor marcies to apporimate the training matrix. In this module, the expected
+        training data is a list of tuples. Every tuple consists of a column index, a row index, and the value at the location specified by the two indexes.
+        For an example of the data structure you can use to encode those tuples, please see <a href='https://github.com/dotnet/machinelearning/pull/1407/files#diff-ce59b50bd87003b0ffb26912fc4a0e65R144'>MatrixElement.
```

Comment:
><a href='https://github.com/dotnet/machinelearning/pull/1407/files#diff-ce59b50bd87003b0ffb26912fc4a0e65R144'>MatrixElement [](start = 92, length = 123)

I am not sure about how I feel pointing our documentation to the tests. @JRAlexander, what do yo u think?

Reply:
We must have code reference for this sentence and that code need to be tested. I believe a doc has to be self-contained. Because I documented that test as detailed as a doc, I'd like to reuse it here.

---
In reply to: [229414267](https://github.com/dotnet/machinelearning/pull/1450#discussion_r229414267) [](ancestors = 229414267)

Reply:
I'd rather have an actual sample that users could use right away.

Reply:
Instead of the link, I would just add this from the test:         // The following variables defines the shape of a matrix. Its shape is _synthesizedMatrixRowCount-by-_synthesizedMatrixColumnCount.
        // The variable _synthesizedMatrixFirstRowIndex indicates the integer that would be mapped to the first row index. If user data uses
        // 0-based indices for rows, _synthesizedMatrixFirstRowIndex can be set to 0. Similarly, for 1-based indices, _synthesizedMatrixFirstRowIndex
        // could be 1.

Reply:
I believe this is an explanation of what you were trying to show with the test, correct?

Reply:
It's about data preparation, not directly related the trainer.

---
In reply to: [229475909](https://github.com/dotnet/machinelearning/pull/1450#discussion_r229475909) [](ancestors = 229475909)

Reply:
I think it would be better for the user to include the comment instead of the link.

Reply:
Ok. I will add a real C# data structure and comments here.

---
In reply to: [229483731](https://github.com/dotnet/machinelearning/pull/1450#discussion_r229483731) [](ancestors = 229483731)